perfmetrics/scripts/vm_metrics/vm_metrics.py

```diff
@@ -100,133 +98,179 @@ def _validate_start_end_times(self, start_time_sec, end_time_sec):
-                        instance, period):
-    """Fetches the API response for peak and mean metrics.
+                        instance, period, aligner, reducer, group_fields):
+    """Fetches the API response for the requested metrics.
 
     Args:
       metric_type (str): The type of metric
       start_time_sec (int): Epoch seconds
       end_time_sec (int): Epoch seconds
       instance (str): VM instance name
-      period (float): Period over which the mean and peak values are taken
+      period (float): Period over which the values are aligned
+      aligner(str): Operation to be applied at points of each period
+      reducer(str): Operation to aggregate data points accross multiple metrics
+      group_fields(list[str]): The fields we want to aggregate using reducer
     Returns:
-      list[peak metrics API response, mean metrics API response]
+      metrics API response (object)
     Raises:
       GoogleAPICallError
+
     """
 
     client = monitoring_v3.MetricServiceClient()
     interval = monitoring_v3.TimeInterval(
         end_time={'seconds': int(end_time_sec)},
         start_time={'seconds': int(start_time_sec)})
 
-    aggregation_peak = monitoring_v3.Aggregation(
-        alignment_period={'seconds': period},
-        per_series_aligner=monitoring_v3.Aggregation.Aligner
-        .ALIGN_MAX,
-    )
-
-    aggregation_mean = monitoring_v3.Aggregation(
+    aggregation = monitoring_v3.Aggregation(
         alignment_period={'seconds': period},
-        per_series_aligner=monitoring_v3.Aggregation.Aligner
-        .ALIGN_MEAN,
+        per_series_aligner=getattr(monitoring_v3.Aggregation.Aligner, aligner),
+        cross_series_reducer=getattr(monitoring_v3.Aggregation.Reducer,reducer),
+        group_by_fields=group_fields
     )
-
-    metric_filter = ('metric.type = "{metric_type}" AND '
-                     'metric.label.instance_name ={instance_name}').format(
-                         metric_type=metric_type, instance_name=instance)
-    try:
-      peak_metrics_response = client.list_time_series({
-          'name':PROJECT_NAME,
-          'filter':metric_filter,
-          'interval':interval,
-          'view':monitoring_v3.ListTimeSeriesRequest
-          .TimeSeriesView.FULL,
-          'aggregation':aggregation_peak,
-      })
-    except:
-      raise GoogleAPICallError('The request for peak response of ' +
-                               metric_type + ' failed, Please try again.')
-
+    if(metric_type[0:7]=='compute'):
+      metric_filter = (
+          'metric.type = "{metric_type}" AND metric.label.instance_name '
+          '={instance_name}'
+          ).format(metric_type=metric_type, instance_name=instance)
+
+    elif (metric_type[0:6] == 'custom'):
+      metric_filter = (
+          'metric.type = "{metric_type}" AND metric.labels.opencensus_task = '
+          'ends_with("{instance_name}")'
+          ).format(metric_type=metric_type, instance_name=instance)
+
```

Comment:
can you extract this get_metric_type into a seperate function and also add comment that what 0:7 / 0:6 mean?

Reply:
done